foundry-extensions/impl/src/main/java/org/pentaho/ctools/cde/impl/DashboardsImpl.java

```diff
@@ -52,19 +61,41 @@ private void init() {
   public JSONArray getDashboardList( int maxDepth, boolean showHiddenFiles ) {
 
     List<IBasicFile> fileList = readAccess.listFiles( path, getFilterForCDEDashboardFiles(),
-        maxDepth, false, showHiddenFiles );
+      maxDepth, false, showHiddenFiles );
 
     JSONArray dashboardArray = null;
 
     if ( fileList != null && fileList.size() > 0 ) {
       dashboardArray = new JSONArray();
       JSONObject jsonObject;
+      IBasicFileExtended fileExtended;
       try {
         for ( IBasicFile file : fileList ) {
-          jsonObject = new JSONObject();
-          jsonObject.put( "path", file.getFullPath() );
-          jsonObject.put( "name", getNameWithoutExtension( file ) );
-          dashboardArray.put( jsonObject );
+          if ( !file.getFullPath().endsWith( IGNORE_END_PATTERN ) ) {
+            jsonObject = new JSONObject();
+            jsonObject.put( "path", file.getFullPath() );
+            jsonObject.put( "name", getNameWithoutExtension( file ) );
+
+            /**
+             * Using cpf-pentaho-rca implementation which we know
+             * it returns IBasicFileExtended instead of IBasicFile
+             * BACKLOG - 25778
+             */
```

Comment:
Maybe put this part surrounded by something like:
if ( file instanceof IBasicFileExtended ) {
...
}